netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -334,16 +335,18 @@ protected ClientTransportFactory buildTransportFactory() {
     TransportCreationParamsFilterFactory transportCreationParamsFilterFactory =
         dynamicParamsFactory;
     if (transportCreationParamsFilterFactory == null) {
-      SslContext localSslContext = sslContext;
-      if (negotiationType == NegotiationType.TLS && localSslContext == null) {
-        try {
-          localSslContext = GrpcSslContexts.forClient().build();
-        } catch (SSLException ex) {
-          throw new RuntimeException(ex);
+      ProtocolNegotiator negotiator = this.protocolNegotiator;
+      if (negotiator == null) {
+        SslContext localSslContext = sslContext;
+        if (negotiationType == NegotiationType.TLS && localSslContext == null) {
+          try {
+            localSslContext = GrpcSslContexts.forClient().build();
+          } catch (SSLException ex) {
+            throw new RuntimeException(ex);
+          }
         }
+        negotiator = createProtocolNegotiatorByType(negotiationType, localSslContext);
```

Comment:
If `this.protocolNegotiator` is not null won't this overwrite `negotiator` ?

Reply:
It can't be non-null, as this is run only `if (negotiator == null)`.

Reply:
oh, one indentation level out.  my mystake.